### ISSUE TITLE
[Enhancement]: Update "Trash Category" UI to allow for restoration

### ISSUE THREAD

**Issue by kimclow** [woothemes/woocommerce#36088]
### Describe the solution you'd like

In WordPress Core, when you delete a page or a post, the content moves to a "trashed" folder where the user can restore it if required. 

It would be helpful if this same UI were used for product categories. 

We recently had a situation where a customer's homepage pulled in product categories to populate the page. The categories were deleted in error in WP Admin, and we had to do a full backup to restore the site. It would have been really helpful to rather undo the "trash" in WP-Admin. 

### Describe alternatives you've considered

The alternative solution we used was that due to the site having Jetpack installed, we did a full backup to the previous day. 
This solution worked well however it did mean we needed to redo the content changes that had been done in the time between.

**Comment by verofasulo** [comment]
That's a very good point. 

Product categories in WooCommerce follow the WordPress taxonomy model, where deleting a category permanently removes it. I don’t think this can be treated as a WooCommerce-only improvement, since it would require changing the underlying taxonomy behavior.

@gigitux @kmanijak what do you think?

**Comment by gigitux** [comment]
Yes. I think that we can close this issue. Currently, we show an alert that we mention that the category will be "permanently delete".

https://github.com/user-attachments/assets/6afa702e-148d-41f3-8403-e0ffdc570508
